subsys/zbus/zbus.c

```diff
@@ -366,8 +506,22 @@ int zbus_obs_is_chan_notification_masked(const struct zbus_observer *obs,
 
 		if (observation->obs == obs) {
 			*masked = observation_mask->enabled;
+
+			_zbus_update_all_channels_hop(obs);
+
 			return 0;
 		}
 	}
 	return -ESRCH;
 }
+
+int zbus_obs_set_enable(struct zbus_observer *obs, bool enabled)
+{
+	_ZBUS_ASSERT(obs != NULL, "obs is required");
+
+	obs->data->enabled = enabled;
+
+	_zbus_update_all_channels_hop(obs);
```

Comment:
Similarly wants locking here: what happens if someone crosses between the enabled set and the update?

Reply:
All the enable/disable actions finish with an update. If someone crosses an update process and sets the enable over again, a second update will happen just after the first one finishes, and the zbus will consider the latest value during the channel's HOP (Highest Observer Priority). The `_zbus_update_all_channels_hop` will lock the channel access to the `highest_observer_priority` field. So, the sensitive part is spinlocked. 

Am I being superficial in my analyses of the situation?

Reply:
@andyross, please take a look at the `_zbus_chan_update_hop` that is the heart of the HLP protocol. The `_zbus_update_all_channels_hop(obs)` calls the `_zbus_chan_update_hop` for each channel that has `obs` as an observer.

Reply:
Should the channel update part be more protected, or is it okay already?